### DISCUSSION TITLE
[Suggestion/Bug?]: Review Camera's recording padding and other cameras column smushed/no scrolling

### DISCUSSION THREAD

**Discussion by bakedhorse** [blakeblackshear/frigate#12030] [Report a Bug] [answered]
### Describe the problem you are having

I like the new Reviewer for viewing motion, alert, and detections. However some of the padding seems too much, and views don't seem to work as intended(?)

The first image has a lot of empty space, around the view itself and below the row of cameras. Notably the row of cameras is scroll-able.
The second image looks fine, maybe a little too much padding around the view but not as bad as the first image. However the column of cameras is smushing all the cameras on-screen at once with no scrolling. Causing the views of the other cameras to be tiny with large black borders/sides to them. 

### Steps to reproduce

1. Setup frigate with a lot of cameras (12 cameras in my case)
2. Let them record for awhile
3. Look at any motion/detection with a specific camera.

### Version

0.14.0-5b60785

### Frigate config file

```yaml
Not relevant.
```

### Relevant log output

```shell
None.
```

### Operating system

Debian

### Install method

Docker Compose

### Network connection

Mixed

### Camera make and model

Not relevant.

### Any other information that may be helpful

![Screenshot_20240617_133615](https://github.com/blakeblackshear/frigate/assets/101318210/05a4b04d-a240-4c4d-8688-fbad680e8d51)
![Screenshot_20240617_133514](https://github.com/blakeblackshear/frigate/assets/101318210/a3d103e0-2dae-41eb-aa59-62b5d447d22c)

**Comment by NickM-27** [comment]
both of these are going to depend on the size of the window, it uses the size to decide if things should be scrollable and how to fit into the layout correctly. I am not able to reproduce either of these issues, might be good to know if resizing fixes it or if things stay the same

**Reply by bakedhorse** [reply]
It doesn't occur on mobile. The column bug happened in Brave (with the shields off) and Chrome, but seemingly not Firefox. I've tried resizing and it doesn't fix it. Though sometimes the column seems to be scroll-able when I click on the "History" button. But it bugs when clicking any detection on the Review page.

The padding is still off for both Firefox and Brave. I've tried resizing the window and it doesn't seem to improve too much. Making the window smaller makes it fill in more, but making the window wider makes the view smaller. Full screen doesn't fix it much either.

**Reply by bakedhorse** [reply]
On another note, the screenshots were taken with a normal maximized web browser.  So it's *shouldn't* be an odd size for it. A normal 16:9 1080p monitor.

**Reply by NickM-27** [reply]
the layout is entirely different on mobile so that is not really relevant. 

It is possible your cameras are reporting a weird size in the browser, in all testing in chrome and firefox the 16:9 video takes up the entire space

![Screen Shot 2024-06-17 at 13 17 30 PM](https://github.com/blakeblackshear/frigate/assets/14866235/346b98f9-d4e6-4820-b142-238839a36f68)

**Reply by bakedhorse** [reply]
My cameras are using high resolutions, but nothing too odd. So maybe? I do have 1080p cameras that have the same padding problem.
Here's all the resolutions I can find. They're all amcrest cameras.

1. 2592x1944 (4:3 causes the vertical column view) (Model: IP5M-T1179E)
2. 3840x2160 (16:9 QFHD, has the horizontal row view) (Model: IP8M-T2599E)
3. 1920x1080 (16:9, has horizontal row view) (Model: IP2M-841B-V3)

**Reply by NickM-27** [reply]
usually it can be something where the video has odd encoding causing the browser to interpret it wrong. Using dahuas means it most likely is not the issue. What are the detect resolutions for these?

**Reply by bakedhorse** [reply]
They're all 640x480

**Reply by NickM-27** [reply]
Okay, for the 16:9 cameras I would suggest using a detect resolution that matches the cameras aspect ratio. For example 860x484, you can set this on the detect config and frigate will handle the resizing

**Reply by bakedhorse** [reply]
Okay that seems to have fixed the padding problem. It's not exactly ideal but it works.
The column bug still occurs.

**Reply by bakedhorse** [reply]
The padding has been fine for the past hour after changing the detect streams to be 16:9. Problem is now that I would need to re-do my masks and zones. And it now takes a little more resources from ffmpeg resizing the detect stream. The alternative is changing the camera streams' resolution but that increases bandwidth. And not all my cameras are capable of a 16:9 substream.
The extra resources usage is negligible but also unnecessary.   

I hope this isn't the permanent fix.

**Reply by NickM-27** [reply]
assuming that you have setup your masks and zones in 0.14 then that shouldn't be necessary as they are stored as relative coordinates

**Reply by NickM-27** [reply]
you could also run at 640x360, I was erring on the side of quality over resource usage

**Reply by bakedhorse** [reply]
> assuming that you have setup your masks and zones in 0.14 then that shouldn't be necessary as they are stored as relative coordinates

*Thank Frigate Developers*

> you could also run at 640x360, I was erring on the side of quality over resource usage

I can see about doing that. I'm trying to balance it between both quality and resources. Don't want to make the already low res image into a smudge

**Reply by bakedhorse** [reply, marked as answer]
Since this is fixed, *though in a band-aid way imo*, imma just close this for now.

Fix is having your detect stream have the same aspect ratio as the live/main stream. *The squashed camera column bug is still happening but it seems to be a chromium bug as it doesn't occur to firefox.*
